From b149ab84f392fc5ddfb89dc040f80f20a8635a1a Mon Sep 17 00:00:00 2001
From: Jianxiong Pan <jianxiong.pan@amlogic.com>
Date: Thu, 24 Mar 2022 22:22:56 +0800
Subject: [PATCH 29/95] mm: add watch point driver. [1/2]

PD#SWPL-76021

Problem:
add watch point driver.

Solution:
porting watch point driver.

Verify:
s4_ap222.

Change-Id: I4b14370000587b8d18c80330ba442e6c1bfb3e3c
Signed-off-by: Jianxiong Pan <jianxiong.pan@amlogic.com>
---
 arch/arm/kernel/hw_breakpoint.c   | 4 ++++
 arch/arm64/kernel/hw_breakpoint.c | 7 +++++++
 2 files changed, 11 insertions(+)

diff --git a/arch/arm/kernel/hw_breakpoint.c b/arch/arm/kernel/hw_breakpoint.c
index b1423fb130ea..cf5cf89d3373 100644
--- a/arch/arm/kernel/hw_breakpoint.c
+++ b/arch/arm/kernel/hw_breakpoint.c
@@ -91,7 +91,11 @@ static u8 max_watchpoint_len __ro_after_init;
 	WRITE_WB_REG_CASE(OP2, 14, VAL);	\
 	WRITE_WB_REG_CASE(OP2, 15, VAL)
 
+#ifdef CONFIG_AMLOGIC_WATCHPOINT
+u32 read_wb_reg(int n)
+#else
 static u32 read_wb_reg(int n)
+#endif
 {
 	u32 val = 0;
 
diff --git a/arch/arm64/kernel/hw_breakpoint.c b/arch/arm64/kernel/hw_breakpoint.c
index 2a7f21314cde..29f3803e4b92 100644
--- a/arch/arm64/kernel/hw_breakpoint.c
+++ b/arch/arm64/kernel/hw_breakpoint.c
@@ -102,7 +102,11 @@ int hw_breakpoint_slots(int type)
 	WRITE_WB_REG_CASE(OFF, 14, REG, VAL);	\
 	WRITE_WB_REG_CASE(OFF, 15, REG, VAL)
 
+#ifdef CONFIG_AMLOGIC_WATCHPOINT
+u64 read_wb_reg(int reg, int n)
+#else
 static u64 read_wb_reg(int reg, int n)
+#endif
 {
 	u64 val = 0;
 
@@ -117,7 +121,10 @@ static u64 read_wb_reg(int reg, int n)
 
 	return val;
 }
+
+#ifndef CONFIG_AMLOGIC_WATCHPOINT
 NOKPROBE_SYMBOL(read_wb_reg);
+#endif
 
 static void write_wb_reg(int reg, int n, u64 val)
 {
-- 
2.25.1

